router/router.go

```diff
@@ -85,38 +95,44 @@ func (r *router) handleChange(change swim.Change) {
 
 // Get the client for a certain destination from our internal cache, or
 // delegates the creation to the ClientFactory.
-func (r *router) GetClient(key string) (interface{}, error) {
+func (r *router) GetClient(key string) (client interface{}, isRemote bool, err error) {
 	dest, err := r.ringpop.Lookup(key)
 	if err != nil {
-		return nil, err
+		return
 	}
 
 	r.rw.RLock()
-	client, ok := r.clientCache[dest]
+	cachedEntry, ok := r.clientCache[dest]
 	r.rw.RUnlock()
 	if ok {
-		return client, nil
+		client = cachedEntry.client
+		isRemote = cachedEntry.isRemote
+		return
 	}
 
 	// no match so far, get a complete lock for creation
 	r.rw.Lock()
 	defer r.rw.Unlock()
 
 	// double check it is not created between read and complete lock
-	client, ok = r.clientCache[dest]
+	cachedEntry, ok = r.clientCache[dest]
 	if ok {
-		return client, nil
+		client = cachedEntry.client
+		isRemote = cachedEntry.isRemote
+		return
```

Comment:
Naked returns are discouraged for bigger functions. https://tour.golang.org/basics/7